src/librustc_mir/build/matches/mod.rs

```diff
@@ -1290,18 +1290,18 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
         // Insert a Shallow borrow of the prefixes of any fake borrows.
         for place in fake_borrows
         {
-            let mut prefix_cursor = &place.projection;
-            while let Some(box Projection { base, elem }) = prefix_cursor {
+            for (i, elem) in place.projection.iter().enumerate().rev() {
+                let proj_base = &place.projection[..i];
```

Comment:
This `for` loop + `let proj_base` pattern occurs a lot; you might want to create a helper for it.

Reply:
For the record https://github.com/rust-lang/rust/pull/64518. This can be done in cases like the one above, where we iterate in reverse order, when going forward is not possible. At the same time the forward case is more obvious to see that being on element i, you need `0..i`. Anyway, a way to iterate over the current elem but carrying the rest in a slice could be nice. Unsure if such a thing exist.

/cc @Centril @oli-obk